src/responder/kcm/kcmsrv_ccache.c

```diff
@@ -49,17 +49,17 @@ errno_t kcm_cc_new(TALLOC_CTX *mem_ctx,
     krb5_error_code kret;
     errno_t ret;
 
-    cc = talloc_zero(mem_ctx, struct kcm_ccache);
-    if (cc == NULL) {
-        return ENOMEM;
-    }
-
     ret = kcm_check_name(name, owner);
     if (ret != EOK) {
         DEBUG(SSSDBG_CRIT_FAILURE, "Name %s is malformed\n", name);
         return ret;
     }
 
+    cc = talloc_zero(mem_ctx, struct kcm_ccache);
+    if (cc == NULL) {
+        return ENOMEM;
+    }
+
```

Comment:
So, personally I'd have taken at least two different approaches than yours.
- Just `talloc_free(cc);` in case the name is malformed;
- Have a goto and a check in the end of the function;

However, I don't have a strong preference and the patch looks good as it is.

Reply:
suggestion taken, I used the goto approach because it will make my life easier with another patch in the series